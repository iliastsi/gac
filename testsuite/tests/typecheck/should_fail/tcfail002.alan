condition() : proc
  foo : byte[16];
{
  if(3<'a')
    writeString("Oops\n");
  if("foo" == "yes")
    writeString("Oops\n");

  foo = "I got " + "42\n";
}
