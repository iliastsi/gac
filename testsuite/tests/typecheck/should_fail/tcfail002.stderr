tcfail002.alan:4:6: Error: Type mismatch at `3 < 'a''
	First argument of `<' is of type `int'
	Second argument of `<' is of type `byte'

tcfail002.alan:6:6: Error: Type mismatch at `"foo" == "yes"'
	Expected `int' or `byte' but expression `"foo"' is of type `array of byte'

tcfail002.alan:9:9: Error: Type mismatch at `"I got " + "42\n"'
	Expected `int' or `byte' but expression `"I got "' is of type `array of byte'

